lib/spack/llnl/util/filesystem.py

```
@@ -113,6 +113,59 @@ def path_contains_subdirectory(path, root):
     return norm_path.startswith(norm_root)
 
 
+@memoized
+def file_command(*args):
+    file_cmd = which("file")
+    for arg in args:
+        file_cmd.add_default_arg(arg)
+    return file_cmd
```

If `which` returns None (it would on Windows right?) then this function should raise an exception.

---

Addressed.